### PR TITLE
Drop columns after loading samples in prepare_tf_dataset

### PR DESCRIPTION
**Pull request by Rocketknight1**
Another super-small fix to `prepare_tf_dataset()` - this time we apply the same fix we applied to `to_tf_dataset()`, and keep columns until after samples have been loaded from the dataset. This ensures that columns that are needed to compute the transform aren't dropped.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
I think this PR is ready to go, but it's waiting on another PR to be merged and a release in Datasets. Tests will fail until that makes it through, so I won't merge until then!

**Review by gante — APPROVED** [review]
LGTM 👍

**Review by gante — APPROVED** [review]
👍

**Pull request merged**
